models/articles.models.js

```diff
@@ -10,3 +10,24 @@ exports.fetchArticles = () => {
   `;
   return db.query(queryString).then((result) => result.rows);
 };
+
+exports.fetchArticleById = (article_id) => {
+
+  if (!/^\d+$/.test(article_id)) {
+    return Promise.reject({ status: 400, msg: "Invalid id" });
```

Comment:
This error could be handled for you by PSQL if you let the invalid id get to the query. You'd get a psql error which you could then handle independently. This approach also works, but I wanted to make sure you were aware of the alternative.
The only problem with this is that your controller might end up bloated - bat for now it's totally fine.

Reply:
Refactored to use psql error handler